Issue building Vitess binaries on Mac OSX

#### Overview of the Issue
Encountered a new error attempting to build Vitess binaries (really just `vtctlclient` + `vtexplain`) on Mac OSX. 
FWIW, this is a relatively (within the last year) new issue as this used to compile without any problems. 

This has been fixed upstream; confirmed [upgrading moby/term](https://github.com/vitessio/vitess/pull/7787) fixes this issue. 

#### Log Fragments

```
Darwin tw-mbp-gedgar 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64
```

```
Mon Apr 5 19:28:14 PDT 2021: Building source tree
# github.com/moby/term
../../../pkg/mod/github.com/moby/term@v0.0.0-20200429084858-129dac9f73f6/tc.go:13:28: undefined: unix.SYS_IOCTL
../../../pkg/mod/github.com/moby/term@v0.0.0-20200429084858-129dac9f73f6/tc.go:18:28: undefined: unix.SYS_IOCTL
../../../pkg/mod/github.com/moby/term@v0.0.0-20200429084858-129dac9f73f6/termios_bsd.go:24:31: undefined: unix.SYS_IOCTL
../../../pkg/mod/github.com/moby/term@v0.0.0-20200429084858-129dac9f73f6/termios_bsd.go:37:31: undefined: unix.SYS_IOCTL
```

--- a/go.mod
+++ b/go.mod
@@ -64,6 +64,7 @@ require (
 	github.com/minio/minio-go v0.0.0-20190131015406-c8a261de75c1
 	github.com/mitchellh/go-testing-interface v1.14.0 // indirect
 	github.com/mitchellh/mapstructure v1.2.3 // indirect
+	github.com/moby/term v0.0.0-20201216013528-df9cb8a40635 // indirect
 	github.com/montanaflynn/stats v0.6.3
 	github.com/olekukonko/tablewriter v0.0.5-0.20200416053754-163badb3bac6
 	github.com/onsi/ginkgo v1.10.3 // indirect

